spec/commands/feed/fetch_one_spec.rb

```diff
@@ -1,94 +1,82 @@
 # frozen_string_literal: true
 
 RSpec.describe Feed::FetchOne do
-  let(:daring_fireball) do
-    double(
-      id: 1,
-      url: "http://daringfireball.com/feed",
-      last_fetched: Time.zone.local(2013, 1, 1),
-      stories: []
-    )
+  def create_entry(opts: {})
+    entry = {
+      comments: "https://news.ycombinator.com/item?id=38246668",
+      published: Time.zone.now,
+      summary: "<a href=\"https://news.ycombinator.com/item?id=38246682\">Comments</a>",
+      title: "Run LLMs on my own Mac fast and efficient",
+      url: "https://www.secondstate.io/articles/fast-llm-inference",
+      content: "",
+      id: "test"
+    }.merge(opts)
+    double(entry)
   end
 
-  before { stub_request(:get, "http://daringfireball.com/feed") }
+  def create_raw_feed(last_modified: Time.zone.now, entries: [])
+    double(last_modified:, entries:)
+  end
 
   context "when no new posts have been added" do
     it "does not add any new posts" do
-      expect(FeedRepository).to receive(:set_status)
-      fake_feed = double(last_modified: Time.zone.local(2012, 12, 31))
-      expect(Feedjira).to receive(:parse).and_return(fake_feed)
+      feed = build(:feed, last_fetched: Time.zone.local(2013, 1, 1))
+      raw_feed = double(last_modified: Time.zone.local(2012, 12, 31))
 
       allow(Feed::FindNewStories).to receive(:call).and_return([])
 
-      expect(StoryRepository).not_to receive(:add)
+      allow(described_class).to receive(:fetch_raw_feed).and_return(raw_feed)
+      expect(FeedRepository).to receive(:set_status).with(:green, feed)
 
-      described_class.call(daring_fireball)
+      expect { described_class.call(feed) }.not_to change(feed.stories, :count)
     end
   end
 
   context "when new posts have been added" do
-    let(:now) { Time.zone.now }
-    let(:new_story) { double }
-    let(:old_story) { double }
-
-    let(:fake_feed) do
-      double(last_modified: now, entries: [new_story, old_story])
-    end
+    it "only adds posts that are new" do
+      feed = create(:feed)
+      opts = { published: Time.zone.local(2013, 1, 1) }
+      raw_feed = create_raw_feed(entries: [create_entry, create_entry(opts:)])
```

Comment:
I might split this into two tests, one for each type of entry:

```rb
it "adds new posts"
it "does not add old posts"
```

Aside from making test cases more explicit, that would also give us more options for wrapping with the 5 line limit.